packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -301,66 +275,71 @@ export async function sideloadAddIn(manifestPath: string, app?: OfficeApp, canPr
     appType = AppType.Desktop;
   }
 
-  const isDesktop: boolean = appType === AppType.Desktop ? true : false;
-  let sideloadFile: string | undefined;
   const manifest: ManifestInfo = await readManifestFile(manifestPath);
   const appsInManifest = getOfficeAppsForManifestHosts(manifest.hosts);
   const isTest: boolean = process.env.WEB_SIDELOAD_TEST !== undefined;
 
-  if (!isDesktop && document === undefined) {
-    throw new Error(`For sideload to web, you need to specify a document url.`);
-  }
-  
-  if (isDesktop) {
-    if (app) {
-      if (appsInManifest.indexOf(app) < 0) {
-        throw new Error(`The Office Add-in manifest does not support ${getOfficeAppName(app)}.`);
-      }
-    } else {
-      switch (appsInManifest.length) {
-        case 0:
-          throw new Error("The manifest does not support any Office apps.");
-        case 1:
-          app = appsInManifest[0];
-          break;
-        default:
-          if (canPrompt) {
-            app = await chooseOfficeApp(appsInManifest);
-          } else {
-            throw new Error("Please specify the Office app.");
-          }
-          break;
-      }
+  if (app) {
```

Comment:
For sideload to web, it will prompt if the manifest contains more than one host. This is because we cannot tell the Office app from the document url, and not all Office app are supported. Most projects only support one host so it usually won't prompt. The --app parameter can be specified to avoid any prompts.